oktapam/fwprovider/resource_server_checkout_settings.go

```diff
@@ -0,0 +1,277 @@
+				MarkdownDescription: "If provided, only the account identifiers listed are excluded from the checkout requirement. This list is only considered if `checkout_required` is set to `true`. Only one of `include_list` and `exclude_list` can be specified in a request since they are mutually exclusive.",
+			},
+			"include_list": schema.ListAttribute{
+				ElementType:         types.StringType,
+				Optional:            true,
+				Description:         "If provided, only the account identifiers listed are required to perform a checkout to access the resource. This list is only considered if `checkout_required` is set to `true`. Only one of `include_list` and `exclude_list` can be specified in a request since they are mutually exclusive.",
+				MarkdownDescription: "If provided, only the account identifiers listed are required to perform a checkout to access the resource. This list is only considered if `checkout_required` is set to `true`. Only one of `include_list` and `exclude_list` can be specified in a request since they are mutually exclusive.",
+			},
+			"project": schema.StringAttribute{
+				Required:            true,
+				Description:         "The UUID of a Project",
+				MarkdownDescription: "The UUID of a Project",
+			},
+			"resource_group": schema.StringAttribute{
+				Required:            true,
+				Description:         "The UUID of a Resource Group",
+				MarkdownDescription: "The UUID of a Resource Group",
+			},
+		},
+	}
+}
+
+func (r *serverCheckoutSettingsResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	// Retrieve the server checkout settings values from the plan
+	var plan serverCheckoutSettingsResourceModel
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Call the SDK client to creat the server checkout settings from plan
+	serverCheckoutSettings := &pam.ResourceCheckoutSettings{
+		CheckoutRequired:          plan.CheckoutRequired,
+		CheckoutDurationInSeconds: plan.CheckoutDurationInSeconds,
+		IncludeList:               plan.IncludeList,
+		ExcludeList:               plan.ExcludeList,
+	}
+
+	_, err := r.client.SDKClient.ProjectsAPI.UpdateResourceGroupServerBasedProjectCheckoutSettings(ctx, r.client.Team, plan.ResourceGroup, plan.Project).ResourceCheckoutSettings(*serverCheckoutSettings).Execute()
+
+	if err != nil {
+		resp.Diagnostics.AddError("Error creating server checkout settings", err.Error())
+		return
+	}
+	// Set state to fully populated data
+	plan.Id = types.StringValue(formatServerCheckoutSettingsID(plan.ResourceGroup, plan.Project))
+	diags = resp.State.Set(ctx, plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
```

Comment:
What is the purpose of doing this? Errors are already added in resp.Diagnostics

Reply:
I think better to call read after creating the resource so that state will be set based on the returned data. We are following this pattern in other resources too.

Reply:
Removed the redundant check.

I didn't find any mention in the official document suggesting to perform the Read operation inside the Create. I came across an interesting issue related to this topic. https://github.com/hashicorp/terraform-plugin-framework/issues/277. I believe we can eliminate the need for a Read operation after Create/Update with the new Terraform provider framework.